### ISSUE TITLE
Improve docs

### ISSUE THREAD

**Issue by jarandaf** [moebiolabs/moebio_framework#200]
First of all, kudos for such a piece of work, it looks very promising. It's really appreciated!

I've been playing around for a while and I think the documentation could be improved, specially since Javascript is an untyped language. While checking the examples (and docs!) I felt lost many times, since I noticed many times some function arguments are optional and that's not specified anywhere (don't know whether there are default argument values in some cases either).

Also, it's kind of annoying clicking some link from the left panel since the left scroll is restarted (related concepts are grouped an you have to scroll down again and again every time). I am not sure you could do something about that since I understand the documentation static pages are auto-generated. 

Summing up, I think the following would improve the documentation:

* Optional function parameters should be marked as optional.
* Default function argument values should be specified.

**Comment by moebio** [comment]
Thanks @jarandaf The documentation could and will be improved. We'll take into account your comments. In particular, obligatory and optional parameters will be differentiated.
